sdk/tables/aztable/connection_string_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztable
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConnectionStringParser(t *testing.T) {
+	require := require.New(t)
+
+	connStr := "DefaultEndpointsProtocol=https;AccountName=dummyaccount;AccountKey=secretkeykey;EndpointSuffix=core.windows.net"
+	serviceURL, cred, err := parseConnectionString(connStr)
+	require.NoError(err)
+	require.Equal(serviceURL, "https://dummyaccount.table.core.windows.net")
+	require.NotNil(cred)
+}
+
+func TestConnectionStringParserHTTP(t *testing.T) {
+	require := require.New(t)
+
+	connStr := "DefaultEndpointsProtocol=http;AccountName=dummyaccount;AccountKey=secretkeykey;EndpointSuffix=core.windows.net"
+	serviceURL, cred, err := parseConnectionString(connStr)
+	require.NoError(err)
+	require.Equal(serviceURL, "http://dummyaccount.table.core.windows.net")
+	require.NotNil(cred)
```

Comment:
Why aren't we testing any of the fields in 'cred'?

Reply:
The method returns an instance of `azcore.Credential` which is an interface that does not define any fields other than an `AuthenticationPolicy` method. I could change this to return an instance of SharedKeyCredential but in the future this could also return an instance of a SharedAccessSignatureCredential which puts me in a bit of a pickle.

Reply:
I think it's okay for a unit test to use internal details to be more thorough, btw. So in this case it's perfectly fine (in a unit test in the same package) to do the cast to a concrete type.

So:

```go
value, ok := cred.(*SharedKeyCredential)
require.True(ok)  // ie, it's really of this type

// check fields of 'value' which is now of type *SharedKeyCredential
```